onnxruntime/test/contrib_ops/col2im_test.cc

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <stdexcept>
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+#include "core/util/math.h"
+
+namespace onnxruntime {
+namespace test {
+
+template <typename T>
+std::vector<T> _transpose_serialized_vector(std::vector<T> &input, size_t N, size_t C, size_t H, size_t W)
+{
+    size_t input_size = input.size();
+    if (input_size == 0){
+        throw std::runtime_error("Invalid input");
+    }
+    std::vector<T> trans_vec(input);
+
+    std::cout << "input: (";
+    for(size_t i = 0; i < input_size; ++i)
+      std::cout << trans_vec[i] << ", ";
+    std::cout << ")" << std::endl;
+
+    for(size_t n = 0; n < N; ++n)
+      for(size_t c = 0; c < C; ++c)
+        for(size_t i = 0; i < H; ++i)
+          for(size_t j = i+1; j < W; ++j)
+              std::swap(trans_vec[n*(C*H*W) + c*(H*W) + (H*i + j)], trans_vec[n*(C*H*W) + c*(H*W) + (W*j + i)]);
```

Comment:
It doesn't affect these test-cases since H=W here, but above line seems incorrect. If the order is NCHW, then the last part should be `H*j + i`, right? You swap only i and j, not the strides.

Reply:
good catch. Added a unit test with 4x3 image to guard this agains regression